File: Hackerrank_30daysOFcode_Go/Day_09.go
package main

import "fmt"

func factorial(n int) int {
  
  if(n==0){

      return 1

  }

  if (n == 1) {
      
      return 1
  }

  return n * factorial(n-1)
}

func main() {

  var n int
  
  fmt.Scan(&n)


  result := factorial(n)
  
  fmt.Printf("%d", result)
  
}

File: 30daysOFcode_Php/day 09.php
<?php

// Complete the factorial function below.
function factorial(int $n) : int {
    if ($n <= 1) return 1;
    return $n * factorial($n-1);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$stdin = fopen("php://stdin", "r");

fscanf($stdin, "%d\n", $n);

$result = factorial($n);

fwrite($fptr, $result . "\n");

fclose($stdin);
fclose($fptr);
